repos/spack_repo/builtin/packages/py_geopmpy/package.py

```diff
@@ -13,30 +13,48 @@ class PyGeopmpy(PythonPackage):
 
     homepage = "https://geopm.github.io"
     git = "https://github.com/geopm/geopm.git"
-    url = "https://github.com/geopm/geopm/tarball/v3.1.0"
+    url = "https://github.com/geopm/geopm/tarball/v3.2.0"
 
     maintainers("bgeltz", "cmcantalupo")
     license("BSD-3-Clause")
     tags = ["e4s"]
 
     version("develop", branch="dev", get_full_repo=True)
+    version("3.2.0", sha256="b708233e1bfda66408c500f2ac0cbaf042140870bffdced12dd7cabbd18e0025")
     version("3.1.0", sha256="2d890cad906fd2008dc57f4e06537695d4a027e1dc1ed92feed4d81bb1a1449e")
 
+    for ver in ["3.1.0", "3.2.0", "develop"]:
+        depends_on(f"py-geopmdpy@{ver}", type="run", when=f"@{ver}")
+        depends_on(f"geopm-runtime@{ver}", type=("build", "run"), when=f"@{ver}")
+
     depends_on("python@3.6:3", type=("build", "run"))
-    depends_on("py-setuptools@53.0.0:", type="build")
+    depends_on("py-setuptools@53.0.0:", when="@3.1", type="build")
+    depends_on("py-setuptools@59.6.0:", when="@3.2:", type="build")
     depends_on("py-setuptools-scm@7.0.3:", when="@3.1:", type="build")
     depends_on("py-build@0.9.0:", when="@3.1:", type="build")
-    depends_on("py-cffi@1.14.5:", type="run")
+    depends_on("py-cffi@1.14.5:", when="@3.1.0", type="run")
+    depends_on("py-cffi@1.14.5:", when="@3.2:", type=("build", "run"))
     depends_on("py-natsort@8.2.0:", type="run")
     depends_on("py-numpy@1.19.5:", type="run")
```

Comment:
Looks like this should be capped at least for 3.2.0:  `py-numpy@1.19.5:1`.

Per https://github.com/geopm/geopm/blob/v3.2.0/geopmpy/pyproject.toml:

```
dependencies = ['cffi>=1.13.2',
                'cycler>=0.11.0',
                'natsort>=8.1.0',
                'numpy>=1.19.5,<2.0.0',
                'pandas>=1.1.5',
                'psutil>=5.4.8',
                'pyyaml>=6.0.0',
                'tables>=3.7.0',
                'geopmdpy',
]
```